### PR TITLE
add delete require and did some main page

### PR DESCRIPTION
**Pull request by Olesya987**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -9,37 +9,43 @@ import medical from "./source/images/medical-2.png";
...
 function App() {
-  const history = useHistory();
-  const [authReg, setAuthReg] = useState({
-    text: "Вход в систему",
-    flag: false,
-    login: "",
-    token: "",
-  });
+  const [authReg, setAuthReg] = useState(
+    JSON.parse(localStorage.getItem("info")) || {
```
Comment (usatovaDasha) [inline src/App.js]: why you need info in storage?

File: src/App.js [hunk 2]
```diff
@@ -9,37 +9,43 @@ import medical from "./source/images/medical-2.png";
...
                 color="primary"
                 type="none"
                 onClick={() => {
-                  setAuthReg({
-                    text: "Вход в систему",
-                    flag: false,
-                    login: "",
-                    token: "",
-                  });
+                  localStorage.setItem(
```
Comment (usatovaDasha) [inline src/App.js]: in function

File: src/AppGrid/AppGrid.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  Button,
+} from "@material-ui/core";
+import DeleteIcon from "@material-ui/icons/Delete";
+import EditIcon from "@material-ui/icons/Edit";
+import "./AppGrid.scss";
+
+const AppGrid = ({ setData, data }) => {
```
Comment (usatovaDasha) [inline src/AppGrid/AppGrid.js]: App component, please give correct name

File: src/AppGrid/AppGrid.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  const [characters, setCharacters] = useState([]);
```
Comment (usatovaDasha) [inline src/AppGrid/AppGrid.js:20]: you need 2 characters array?

File: src/AppGrid/AppGrid.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    if (data.length === 0) {
+      const info = JSON.parse(localStorage.getItem("info"));
+      axios
+        .get("http://localhost:8080/appointment/get", {
+          headers: { authorization: info.token },
+        })
+        .then((res) => {
+          const temp = res.data.appointments;
+          const arr = [];
+          for (let key in temp[0]) {
```
Comment (usatovaDasha) [inline src/AppGrid/AppGrid.js]: if res.data.appointments will length 0?

File: src/AppGrid/AppGrid.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+                    <h2>{value}</h2>
+                  </TableCell>
+                ))}
+                <TableCell> </TableCell>
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              {data.map((row) => (
+                <TableRow hover key={row._id}>
+                  {characters.map((name, index) => (
```
Comment (usatovaDasha) [inline src/AppGrid/AppGrid.js:112]: if the order of the properties is different than in the first array?

File: src/AppGrid/AppGrid.js [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+                      </IconButton>
+                    </div>
+                  </TableCell>
+                </TableRow>
+              ))}
+            </TableBody>
+          </Table>
+        </TableContainer>
+      </div>
+      <Dialog
```
Comment (usatovaDasha) [inline src/AppGrid/AppGrid.js]: in component

File: src/AppGrid/AppGrid.js [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+        open={delProps.open}
+        onClose={handleCloseDel}
+        aria-labelledby="alert-dialog-title"
```
Comment (usatovaDasha) [inline src/AppGrid/AppGrid.js]: need this properties?

File: src/AppGrid/AppGrid.js [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+        aria-describedby="alert-dialog-description"
+      >
+        <DialogTitle id="alert-dialog-title">{"Удалить прием"}</DialogTitle>
```
Comment (usatovaDasha) [inline src/AppGrid/AppGrid.js]: no need {""}

File: src/AppGrid/AppGrid.js [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+        <DialogContent>
+          <DialogContentText id="alert-dialog-description">
+            Вы действительно хотите удалить прием?
+          </DialogContentText>
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={handleCloseDel} color="primary" variant="outlined">
```
Comment (usatovaDasha) [inline src/AppGrid/AppGrid.js]: correct call function

File: src/AppGrid/AppGrid.js [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+            Отмена
+          </Button>
+          <Button
+            onClick={delFunc}
```
Comment (usatovaDasha) [inline src/AppGrid/AppGrid.js]: higher

File: src/AppGrid/AppGrid.scss [hunk 12]
```diff
@@ -0,0 +1,40 @@
+.table{
```
Comment (usatovaDasha) [inline src/AppGrid/AppGrid.scss]: spaces to the file

File: src/AppGrid/AppGrid.scss [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+.container-grid{
+  max-height: 640px;
+}
+
+::-webkit-scrollbar {
+  background: rgba(255, 255, 255, 0.1);
+  border-radius: 10px;
+  width: 7px;
+
```
Comment (usatovaDasha) [inline src/AppGrid/AppGrid.scss]: remove line

File: src/Appointments/Appointments.js [hunk 14]
```diff
@@ -1,32 +1,13 @@
 import React, { useState } from "react";
-import { Route, Switch, useHistory, Redirect } from "react-router-dom";
 import Sort from "../Sort/Sort";
+import AppGrid from "../AppGrid/AppGrid";
 import "./Appointments.scss";
```
Comment (usatovaDasha) [inline src/Appointments/Appointments.js:4]: add empty line

File: src/Sort/Sort.js [hunk 15]
```diff
@@ -1,12 +1,177 @@
...
+  Button,
+} from "@material-ui/core";
 import "./Sort.scss";
-const Sort = () => {
-  return (
-  <div className='sort-main'>
-    sort
 
-  </div>
+const Sort = ({ setData }) => {
```
Comment (usatovaDasha) [inline src/Sort/Sort.js]: please correct name

File: src/Sort/Sort.js [hunk 16]
```diff
@@ -1,12 +1,177 @@
...
+                  </InputLabel>
+                  <Select
+                    labelId="demo-simple-select-outlined-label"
+                    id="input-docName"
+                    name="input-docName"
+                    value={require.docName}
+                    onChange={handleChange("docName")}
+                    label="Врач"
+                  >
+                    <MenuItem value="">
```
Comment (usatovaDasha) [inline src/Sort/Sort.js]: map MenuItem

File: src/Sort/Sort.js [hunk 17]
```diff
@@ -1,12 +1,177 @@
...
+                  name="input-date"
+                  label="Дата"
+                  variant="outlined"
+                  type="date"
+                  className="textField"
+                  InputLabelProps={{
+                    shrink: true,
+                  }}
+                  value={require.date}
+                  onChange={handleChange("date")}
```
Comment (usatovaDasha) [inline src/Sort/Sort.js]: correct call function

File: src/Sort/Sort.js [hunk 18]
```diff
@@ -1,12 +1,177 @@
...
+            <Grid key="complaints" item>
+              <div className="auth-div-login">
+                <TextField
+                  id="input-complaints"
+                  name="input-complaints"
+                  label="Жалобы"
+                  type="text"
+                  variant="outlined"
+                  value={require.complaints}
+                  onChange={handleChange("complaints")}
```
Comment (usatovaDasha) [inline src/Sort/Sort.js]: higher

File: src/Sort/Sort.js [hunk 19]
```diff
@@ -1,12 +1,177 @@
...
+            </Grid>
+            <Grid key="add" item>
+              <div className="auth-div-login">
+                {!checkFields() ? (
+                  <Button
+                    className="button-add"
+                    variant="outlined"
+                    color="primary"
+                    type="none"
+                    disabled
```
Comment (usatovaDasha) [inline src/Sort/Sort.js]: disabled={checkFields()}

File: src/Sort/Sort.js [hunk 20]
```diff
@@ -1,12 +1,177 @@
...
+                  type="text"
+                  variant="outlined"
+                  value={require.complaints}
+                  onChange={handleChange("complaints")}
+                />
+              </div>
+            </Grid>
+            <Grid key="add" item>
+              <div className="auth-div-login">
+                {!checkFields() ? (
```
Comment (usatovaDasha) [inline src/Sort/Sort.js]: no need ternary

File: src/Sort/Sort.js [hunk 21]
```diff
@@ -1,12 +1,177 @@
...
+                    Добавить
+                  </Button>
+                )}
+              </div>
+            </Grid>
+          </Grid>
+        </Grid>
+      </Grid>
+    </div>
+    // <div className="sort-main">
```
Comment (usatovaDasha) [inline src/Sort/Sort.js]: remove comments

File: src/Sort/Sort.scss [hunk 22]
```diff
@@ -1,6 +1,47 @@
...
+  flex-grow: 1;
+}
+
+.paper {
+  height: 140px;
+  width: 100px;
+}
+
+.MuiGrid-spacing-xs-1 {
+  width: 100% !important;
```
Comment (usatovaDasha) [inline src/Sort/Sort.scss]: don't use important

File: src/Sort/Sort.scss [hunk 23]
```diff
@@ -1,6 +1,47 @@
...
+}
+
+.textField {
+  margin-left: 8px;
+  margin-right: 8px;
+  width: 200;
+}
+
+.MuiButton-outlined {
+  padding: 15px 15px !important;
```
Comment (usatovaDasha) [inline src/Sort/Sort.scss]: higher

File: src/Add/Add.js [hunk 24]
```diff
@@ -9,9 +9,10 @@ import {
   TextField,
   Button,
 } from "@material-ui/core";
-import "./Sort.scss";
+import "./Add.scss";
```
Comment (usatovaDasha) [inline src/Add/Add.js:12]: imports style

File: src/Add/Add.js [hunk 25]
```diff
@@ -56,6 +57,7 @@ const Sort = ({ setData }) => {
           complaints: "",
         });
         setData(res.data.appointments);
+        setFlag(true);
```
Comment (usatovaDasha) [inline src/Add/Add.js:59]: what it is flag?

File: src/Add/Add.scss [hunk 26]
```diff
@@ -33,15 +33,15 @@
   width: 200;
 }
 
-.MuiButton-outlined {
-  padding: 15px 15px !important;
+.MuiButton-outlined.MuiButtonBase-root {
+  padding: 15px 15px;
```
Comment (usatovaDasha) [inline src/Add/Add.scss]: padding: 15px;

File: src/Add/Add.scss [hunk 27]
```diff
@@ -33,15 +33,15 @@
...
-.formControl {
+.formControl1.MuiFormControl-root {
   margin: 8px;
-  min-width: 229px !important;
+  min-width: 225px;
 }
 
-.MuiInputBase-input{
-  min-width: 229px !important;
+.MuiInputBase-input.MuiOutlinedInput-input{
```
Comment (usatovaDasha) [inline src/Add/Add.scss]: space

File: src/App.js [hunk 28]
```diff
@@ -5,71 +5,66 @@ import Auth from "./Auth/Auth";
...
     JSON.parse(localStorage.getItem("info")) || {
       text: "Вход в систему",
-      flag: false,
       login: "",
       token: "",
     }
   );
 
+  const goToAuth = () => {
+    localStorage.setItem(
```
Comment (usatovaDasha) [inline src/App.js]: const info = {
        text: "Вход в систему",
        login: "",
        token: "",
      };
localStorage.setItem("info", JSON.stringify(info));
setAuthReg(info);

File: src/App.js [hunk 29]
```diff
@@ -5,71 +5,66 @@ import Auth from "./Auth/Auth";
...
+      "info",
+      JSON.stringify({
+        text: "Вход в систему",
+        login: "",
+        token: "",
+      })
+    );
+    setAuthReg(JSON.parse(localStorage.getItem("info")));
```
Comment (usatovaDasha) [inline src/App.js]: why need storage if you using state?

File: src/AppGrid/AppGrid.js [hunk 30]
```diff
@@ -38,11 +41,29 @@ const AppGrid = ({ setData, data }) => {
         .then((res) => {
           const temp = res.data.appointments;
           const arr = [];
-          for (let key in temp[0]) {
-            if (key !== "_id") arr.push(key);
+          if (temp.length !== 0) {
+            for (let key in temp[0]) {
+              if (key !== "_id") arr.push({ value: key, translate: "" });
+            }
+            for (let i = 0; i < arr.length; i++) {
```
Comment (usatovaDasha) [inline src/AppGrid/AppGrid.js:40]: using switch case

File: src/AppGrid/AppGrid.js [hunk 31]
```diff
@@ -38,11 +41,29 @@ const AppGrid = ({ setData, data }) => {
         .then((res) => {
           const temp = res.data.appointments;
           const arr = [];
-          for (let key in temp[0]) {
-            if (key !== "_id") arr.push(key);
+          if (temp.length !== 0) {
+            for (let key in temp[0]) {
+              if (key !== "_id") arr.push({ value: key, translate: "" });
```
Comment (usatovaDasha) [inline src/AppGrid/AppGrid.js]: and immediately add translate

File: src/AppGrid/AppGrid.scss [hunk 32]
```diff
@@ -37,4 +37,12 @@
 .backdrop {
   z-index: 4;
   color: rgb(255, 255, 255);
+}
+
+.center-text{
```
Comment (usatovaDasha) [inline src/AppGrid/AppGrid.scss]: space

File: src/Appointments/Appointments.js [hunk 33]
```diff
@@ -1,13 +1,14 @@
 import React, { useState } from "react";
-import Sort from "../Sort/Sort";
+import Add from "../Add/Add";
 import AppGrid from "../AppGrid/AppGrid";
 import "./Appointments.scss";
 const Appointments = () => {
   const [data, setData] = useState([]);
+  const [flag, setFlag] = useState(true);
```
Comment (usatovaDasha) [inline src/Appointments/Appointments.js:8]: add empty line

File: src/Auth/Auth.js [hunk 34]
```diff
@@ -118,80 +117,83 @@ const Auth = ({ setAuthReg }) => {
...
+          <div className="input-form-auth">
+            <div className="auth-div-login">
+              <TextField
+                id="input-login"
+                name="input-login"
+                label="Login"
+                type="text"
+                variant="outlined"
+                value={values.login}
+                onChange={handleChange("login")}
```
Comment (usatovaDasha) [inline src/Auth/Auth.js]: () => handleChange("login")

File: src/Auth/Auth.js [hunk 35]
```diff
@@ -118,80 +117,83 @@ const Auth = ({ setAuthReg }) => {
...
+            <FormControl className="root margin textField" variant="outlined">
+              <InputLabel htmlFor="outlined-adornment-password">
+                Password
+              </InputLabel>
+              <OutlinedInput
+                id="input-password"
+                name="input-password"
+                type={values.showPassword ? "text" : "password"}
+                value={values.password}
+                onChange={handleChange("password")}
```
Comment (usatovaDasha) [inline src/Auth/Auth.js]: correct call function

File: src/Reg/Reg.js [hunk 36]
```diff
@@ -122,109 +121,112 @@ const Reg = ({ setAuthReg }) => {
...
+          <div className="input-form-reg">
+            <div className="reg-div-login">
+              <TextField
+                id="input-login"
+                name="input-login"
+                label="Login"
+                type="text"
+                variant="outlined"
+                value={values.login}
+                onChange={handleChange("login")}
```
Comment (usatovaDasha) [inline src/Reg/Reg.js]: () => handleChange("login")

File: src/Reg/Reg.js [hunk 37]
```diff
@@ -122,109 +121,112 @@ const Reg = ({ setAuthReg }) => {
...
+                value={values.passwordRepeat}
+                onChange={handleChange("passwordRepeat")}
+                endAdornment={
+                  <InputAdornment position="end">
+                    <IconButton
+                      aria-label="toggle password repeat visibility"
+                      onClick={handleClickShowPasswordRepeat}
+                      edge="end"
+                    >
+                      {values.showPasswordRepeat ? (
```
Comment (usatovaDasha) [inline src/Reg/Reg.js:179]: {values.showPasswordRepeat ? <Visibility /> : <VisibilityOff />}

File: src/ModalDel/ModalDel.js [hunk 38]
```diff
@@ -0,0 +1,67 @@
...
+  };
+
+  return (
+    <Dialog
+      open={open}
+      onClose={() => handleCloseDel()}
+      aria-labelledby="alert-dialog-title"
+      aria-describedby="alert-dialog-description"
+    >
+      <DialogTitle id="alert-dialog-title">{"Удалить прием"}</DialogTitle>
```
Comment (usatovaDasha) [inline src/ModalDel/ModalDel.js]: no need {""}

File: src/Add/Add.js (hunk 24 above)
Comment (Olesya987) [reply src/Add/Add.js:12]: i corrected this comment, but it doesn't show up here

**Pull request merged**